pkg/cmd/env/active.go

```diff
@@ -29,6 +30,25 @@ func activeCmd(cfg *config.Config) *cobra.Command {
 				err                   error
 				availableEnvironments []environment.Environment
 			)
+
+			nonInteractive, err := cmd.Flags().GetBool("non-interactive")
+			if err != nil {
+				nonInteractive = true
+			}
+
+			opts.Streams = userio.NewIOStreamsWithInteractive(
+				cmd.InOrStdin(),
+				cmd.OutOrStdout(),
+				cmd.OutOrStderr(),
+				!nonInteractive,
+			)
+
+			repoParams := []config.Parameter[string]{cfg.Repositories.CPlatform}
+			err = config.Update(cfg.GitHub.Token.Value, opts.Streams, cfg.Repositories.AllowDirty.Value, repoParams)
+			if err != nil {
+				return fmt.Errorf("failed to update config repos: %w", err)
+			}
+
 			cmd.SilenceUsage = true
```

Comment:
The call to `config.Update()` needs an IOStream, which is instantiated on line 39. This in turns needs the value of the non-interactive flag, which is set on line 34. I think this block of code (line 48-50) is executed as early as it could, I don't see a way to execute it earlier than that...

Reply:
the comment was attached to `cmd.SIlenceUsage = true`, that should remain the first line. I don't see this being a problem.

Reply:
Done